mw-webapp/src/logic/wayPage/WayPage.tsx

```diff
@@ -388,101 +377,26 @@ export const WayPage = (props: WayPageProps) => {
-        <div className={styles.goalSubSection}>
-          <Title
-            level={HeadingLevel.h3}
-            text="Goal"
-          />
-          <EditableTextarea
-            text={way.goal.description}
-            onChangeFinish={(description) => changeGoalDescription(way.goal, description)}
-            rows={10}
-            isEditable={isOwner}
-            className={styles.goalDescription}
-          />
-        </div>
-        <div className={styles.goalSubSection}>
-          <HorizontalContainer className={styles.horizontalContainer}>
-            <Title
-              level={HeadingLevel.h3}
-              text="Metrics"
-            />
-            <Tooltip content={`Click to ${wayPageSettings.isGoalMetricsVisible ? "hide" : "open"} goal metrics block`}>
-              <div
-                className={styles.iconContainer}
-                onClick={() => updateWayPageSettings({isGoalMetricsVisible: !wayPageSettings.isGoalMetricsVisible})}
-              >
-                {wayPageSettings.isGoalMetricsVisible ?
-                  <Icon
-                    size={IconSize.MEDIUM}
-                    name="EyeOpenedIcon"
-                  />
-                  :
-                  <Icon
-                    size={IconSize.MEDIUM}
-                    name="EyeSlashedIcon"
-                  />
-                }
-              </div>
-            </Tooltip>
-          </HorizontalContainer>
-          <GoalMetricsBlock
-            isVisible={wayPageSettings.isGoalMetricsVisible}
-            way={way}
-            isEditable={isOwner}
-          />
-        </div>
-        <div className={styles.goalSubSection}>
-          <HorizontalContainer className={styles.horizontalContainer}>
-            <Title
-              level={HeadingLevel.h3}
-              text="Statistics"
-            />
-            <Tooltip content={`Click to ${wayPageSettings.isStatisticsVisible ? "hide" : "open"} statistics block`}>
-              <div
-                className={styles.iconContainer}
-                onClick={() => updateWayPageSettings({isStatisticsVisible: !wayPageSettings.isStatisticsVisible})}
-              >
-                {wayPageSettings.isStatisticsVisible ?
-                  <Icon
-                    size={IconSize.MEDIUM}
-                    name="EyeOpenedIcon"
-                  />
-                  :
-                  <Icon
-                    size={IconSize.MEDIUM}
-                    name="EyeSlashedIcon"
-                  />
-                }
-              </div>
-            </Tooltip>
-          </HorizontalContainer>
-          <WayStatistic
-            dayReports={way.dayReports}
-            wayCreatedAt={way.createdAt}
-            isVisible={wayPageSettings.isStatisticsVisible}
-          />
-        </div>
-      </div>
+        />)}
+
+      <GoalBlock
+        way={way}
+        wayPageSettings={wayPageSettings}
+        updateWaySettings={(updated) => updateWayPageSettings(updated)}
```

Comment:
Use short syntax. Please make notice, its hard to repeat things so much times, please remember it